5-data-modeling/event-manifest-populator/run.py

```diff
@@ -0,0 +1,281 @@
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+import base64
+import sys
+import subprocess
+from datetime import datetime
+
+import argparse
+
+import boto.s3
+from boto.s3.connection import Location
+from boto.s3.key import Key
+
+import boto.emr
+from boto.emr.step import JarStep
+
+
+DIR_WITH_JAR = "./target/scala-2.11/"
+JAR_FILE = "event-manifest-populator-0.1.0-rc1.jar"
+
+S3_REGIONS = {'us-east-1': Location.DEFAULT,
+              'us-west-1': Location.USWest,
+              'us-west-2': Location.USWest2,
+              'eu-west-1': Location.EU,
+              'ap-southeast-1': Location.APSoutheast,
+              'ap-southeast-2': Location.APSoutheast2,
+              'ap-northeast-1': Location.APNortheast,
+              'sa-east-1': Location.SAEast}
+
+S3_LOCATIONS = {v: k for k, v in S3_REGIONS.items()}
+
+SINCE_SHORT_FORMAT = "%Y-%m-%d"
+SINCE_PRECISE_FORMAT = "%Y-%m-%d-%H-%M-%s"
+
+
+def split_full_path(path):
+    """Return pair of bucket without protocol and path
+    Arguments:
+    path - valid S3 path, such as s3://somebucket/events
+    >>> split_full_path('s3://mybucket/path-to-events')
+    ('mybucket', 'path-to-events/')
+    >>> split_full_path('s3://mybucket')
+    ('mybucket', None)
+    """
+    if path.startswith('s3://'):
+        path = path.lstrip('s3://')
+    elif path.startswith('s3n://'):
+        path = path.lstrip('s3n://')
+    else:
+        raise ValueError("S3 path should start with s3:// or s3n:// prefix")
+    parts = path.split('/')
+    bucket = parts[0]
+    path = '/'.join(parts[1:])
+    return (bucket, normalize_prefix(path))
+
+
+def normalize_prefix(path):
+    """Add trailing slash to prefix if it is not present
+    >>> normalize_prefix("somepath")
+    'somepath/'
+    >>> normalize_prefix("somepath/")
+    'somepath/'
+    """
+    if path is None or path is '' or path is '/':
+        return None
+    elif path.endswith('/'):
+        return path
+    else:
+        return path + '/'
+
+
+def run(command):
+    output = subprocess.Popen(command.split(),
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE)
+    (stdout, stderr) = output.communicate()
+    print(stdout)
+    if output.returncode != 0:
+        sys.exit(stderr)
+
+
+def validate_input_path(s3_path):
+    if not s3_path.startswith("s3://"):
+        raise ValueError("Invalid enriched archive path. It should start with s3:// protocol")
+
+
+def validate_since(since):
+    try:
+        datetime.strptime(since, SINCE_SHORT_FORMAT)
+    except ValueError:
+        try:
+            datetime.strptime(since, SINCE_PRECISE_FORMAT)
+        except ValueError:
+            print("Incorrect --since format. Should conform YY-MM-dd or YY-MM-dd-HH-mm-ss format")
+            sys.exit(1)
+
+
+def get_valid_region(location):
+    return S3_LOCATIONS[location]
```

Comment:
In my local install, this dictionary lookup fails with a KeyError because the `location` is a unicode string, but the map contains non-unicode keys...